workspaces/meewee-time-registration/plugins/plugin-meewee-time-registration/meewee-time-registration/src/request/api/azdevops/AzDevOpsClient.ts

```diff
@@ -0,0 +1,131 @@
+import { DiscoveryApi, FetchApi, OAuthApi } from '@backstage/core-plugin-api';
+import { AzDevopsApi, request } from '../../../types/azdevops/types';
+import { ResponseError } from '@backstage/errors';
+
+/** @public */
+export class AzDevOpsClient implements AzDevopsApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly authApi: OAuthApi;
+  private readonly client: FetchApi;
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    authApi: OAuthApi;
+    fetchApi: FetchApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.authApi = options.authApi;
+    this.client = options.fetchApi;
+  }
+
+  private async handleResponse(req: request): Promise<any> {
+    // azure devops scope  '499b84ac-1321-427f-aa17-267ca6975798/.default'
+    const token = await this.authApi.getAccessToken(req.scope);
```

Comment:
There's an existing `azureApiRef` that would be able to get you the tokens needed for this based on how the Adopter has setup their Backstage integration.

Reply:
We removed this one. Sorry, this wasn't removed on the former commits. This is a feature that we're planning to implement in the future.